tests/unit/core/utils/test_compatibility.js

```diff
@@ -702,6 +702,67 @@ describe(__filename, () => {
         reason: INCOMPATIBLE_UNSUPPORTED_PLATFORM,
       });
     });
+
+    // See: https://github.com/mozilla/addons-frontend/issues/6240
+    it('returns incompatible when add-on is non-restartless and FF version >= 61.0', () => {
+      const userAgentInfo = UAParser(userAgentsByPlatform.mac.firefox61);
+      const clientApp = CLIENT_APP_FIREFOX;
+      const addon = createInternalAddon({
+        ...fakeAddon,
+        current_version: {
+          ...fakeAddon.current_version,
+          files: [
+            {
+              ...fakeAddon.current_version.files[0],
+              is_restart_required: true,
+            },
+          ],
+        },
+      });
+
+      expect(
+        getClientCompatibility({
+          addon,
+          clientApp,
+          userAgentInfo,
+        }),
+      ).toEqual({
```

Comment:
This would be more concise and maintainable as `.toMatchObject({ compatible: false, reason: '...' });`